MINOR: Update build.gradle and release.py to upload streams-scala_2.12

rajinisivaram opened the pull request:
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

rajinisivaram committed:
MINOR: Update build.gradle and release.py to upload streams-scala_2.12

ijuma reviewed (commented):
Hmm, why isn't uploadArchivesAll working? We should not need the 2_12 version anymore.

ijuma commented:
For reference, Scala 2.12 was added to the *All commands when we dropped Java 7 support: https://github.com/apache/kafka/commit/e70a191d3038e00790aa95fbd1e16e78c32b79a4#diff-6512f838e273b79676cac5f72456127f

Looks like the update to release.py was incomplete and the supposedly redundant upload archive line was not removed.

rajinisivaram committed:
Address review comment

rajinisivaram commented:
@ijuma Thanks for the review. The various `xxxAll` tasks in `build.gradle` seem to explicitly run tasks for `core`, which is possibly why `streams-scala` didn't get uploaded for `scala 2.12`. I have removed the explicit `2.12` upload in `release.py` and updated `build.gradle` to run tasks for `streams-scala` as well wherever it was running tasks for `core` with different scala versions. Can you take another look? Thanks.

ijuma commented on build.gradle:
```diff
@@ -473,28 +473,30 @@ task reportCoverage(dependsOn: ['jacocoRootReport', 'core:reportCoverage'])
     startParameter = project.getGradle().getStartParameter().newInstance()
     startParameter.projectProperties += [scalaVersion: "${sv}"]
+    // core has a separate source jar for each scala version, streams-scala source is included
+    // in the single streams sources jar
```
@guozhangwang @mjsax The behaviour with regards to the Scala source mentioned in the comment above is a bit weird.

guozhangwang replied:
I think we have a separate jar for each scala version as well for kafka-streams-scala?

rajinisivaram replied:
@ijuma @guozhangwang Sorry, this was an issue with the older version of gradle that I was testing with. Updated that and now it does generate separate source jars (I dont understand why). Updated `build.gradle`.

ijuma commented on build.gradle:
```diff
@@ -525,20 +527,20 @@ def withDefScalaVersions(taskName) {
 
 tasks.create(name: "testConnect", dependsOn: connectPkgs.collect { it + ":test" }) {}
-tasks.create(name: "testAll", dependsOn: withDefScalaVersions('test_core') + pkgs.collect { it + ":test" }) { }
+tasks.create(name: "testAll", dependsOn: withDefScalaVersions('test_scala') + pkgs.collect { it + ":test" }) { }
```
Maybe we should use camel case here and a few lines above for consistency with the other task names.

rajinisivaram replied:
@ijuma Thanks for the review, updated.

ijuma reviewed (approved):
Thanks for the PR, just a nit.

rajinisivaram committed:
Address review comment

ijuma reviewed (approved):
LGTM

guozhangwang commented on release.py:
```diff
@@ -495,7 +495,6 @@ def select_gpg_key():
 if not user_ok("Going to build and upload mvn artifacts based on these settings:\n" + contents + '\nOK (y/n)?: '):
     fail("Retry again later")
 cmd("Building and uploading archives", "./gradlew uploadArchivesAll", cwd=kafka_dir, env=jdk8_env)
-cmd("Building and uploading archives", "./gradlew uploadCoreArchives_2_12 -PscalaVersion=2.12", cwd=kafka_dir, env=jdk8_env)
```
Just curious: seems we do not need this step even before this PR. Is that right?

ijuma replied:
Yes, that should have been removed in a previous PR.

guozhangwang reviewed (commented):
The patch LGTM.

rajinisivaram committed:
Fix srcJarAll for streams-scala

ijuma reviewed (approved):
LGTM

rajinisivaram commented:
retest this please

rajinisivaram commented:
Merging to trunk and 2.0.
